src/tenant/tests/test_views.py

```
@@ -85,101 +92,331 @@ def setUp(self):
+        (redirected back to the request form) before anything is saved."""
+        nonce = DeckRequestService.create_request("John", "Doe", "john.doe@example.com")
+
+        def build_request(deck_name):
+            # distinct deck names so the second form is valid and reaches
+            # form_valid (where the consumed nonce, not a duplicate name, rejects)
+            data = {**self.form_data, "name": deck_name}
+            request = self.factory.post(reverse("tenant:new"), data=data)
+            request.user = AnonymousUser()
+            request.session = {"verified_deck_request": {**data, "nonce": nonce}}
+            # message storage so form_valid's reject path can add a message
+            setattr(request, "_messages", FallbackStorage(request))
```

Ruff (0.15.20)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)